### ISSUE TITLE
xcframework pod lib lint fail

### ISSUE THREAD

**Issue by AdamShi** [CocoaPods/CocoaPods#11532]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

I created an xcframework, this xcframework uses a mix of oc and swift, and dependency on the oc third-party library and the swift third-party library, I want to publish it to the cocoapods , but there are always errors during verification, I use this command for verification
`pod lib lint --verbose --allow-warnings --no-clean`

## What did you expect to happen?

The verification is successful and published to cocoapods

## What happened instead?

```
 Undefined symbols for architecture x86_64:
      "enum case for SwiftEntryKit.EKAttributes.Position.top(SwiftEntryKit.EKAttributes.Position.Type) -> SwiftEntryKit.EKAttributes.Position", referenced from:
          function signature specialization <Arg[7] = Dead> of static SASContractSDK.Overlay.entrySheet(_: __C.UIView, top: Swift.Bool, offset: CoreGraphics.CGFloat, round: CoreGraphics.CGFloat, bgcolor: Swift.Int, dismiss: Swift.Bool, cancelled: (() -> ())?) -> () in SASContractSDK(Overlay.o)
          function signature specialization <Arg[7] = Dead> of static SASContractSDK.Overlay.entrySheetHide(_: __C.UIView, top: Swift.Bool, offset: CoreGraphics.CGFloat, round: CoreGraphics.CGFloat, bgcolor: Swift.Int, dismiss: Swift.Bool, cancelled: (() -> ())?) -> () in SASContractSDK(Overlay.o)
      "enum case for SwiftEntryKit.EKAttributes.Position.bottom(SwiftEntryKit.EKAttributes.Position.Type) -> SwiftEntryKit.EKAttributes.Position", referenced from:
          function signature specialization <Arg[7] = Dead> of static SASContractSDK.Overlay.entrySheet(_: __C.UIView, top: Swift.Bool, offset: CoreGraphics.CGFloat, round: CoreGraphics.CGFloat, bgcolor: Swift.Int, dismiss: Swift.Bool, cancelled: (() -> ())?) -> () in SASContractSDK(Overlay.o)
          function signature specialization <Arg[7] = Dead> of static SASContractSDK.Overlay.entrySheetHide(_: __C.UIView, top: Swift.Bool, offset: CoreGraphics.CGFloat, round: CoreGraphics.CGFloat, bgcolor: Swift.Int, dismiss: Swift.Bool, cancelled: (() -> ())?) -> () in SASContractSDK(Overlay.o)
      "enum case for SwiftEntryKit.EKAttributes.PositionConstraints.SafeArea.overridden(SwiftEntryKit.EKAttributes.PositionConstraints.SafeArea.Type) -> SwiftEntryKit.EKAttributes.PositionConstraints.SafeArea", referenced from:
          function signature specialization <Arg[7] = Dead> of static SASContractSDK.Overlay.entrySheet(_: __C.UIView, top: Swift.Bool, offset: CoreGraphics.CGFloat, round: CoreGraphics.CGFloat, bgcolor: Swift.Int, dismiss: Swift.Bool, cancelled: (() -> ())?) -> () in SASContractSDK(Overlay.o)
          function signature specialization <Arg[7] = Dead> of static SASContractSDK.Overlay.entrySheetHide(_: __C.UIView, top: Swift.Bool, offset: CoreGraphics.CGFloat, round: CoreGraphics.CGFloat, bgcolor: Swift.Int, dismiss: Swift.Bool, cancelled: (() -> ())?) -> () in SASContractSDK(Overlay.o)
      "enum case for SwiftEntryKit.EKAttributes.RoundCorners.bottom(SwiftEntryKit.EKAttributes.RoundCorners.Type) -> (CoreGraphics.CGFloat) -> SwiftEntryKit.EKAttributes.RoundCorners", referenced from:
          function signature specialization <Arg[7] = Dead> of static SASContractSDK.Overlay.entrySheet(_: __C.UIView, top: Swift.Bool, offset: CoreGraphics.CGFloat, round: CoreGraphics.CGFloat, bgcolor: Swift.Int, dismiss: Swift.Bool, cancelled: (() -> ())?) -> () in SASContractSDK(Overlay.o)
          function signature specialization <Arg[7] = Dead> of static SASContractSDK.Overlay.entrySheetHide(_: __C.UIView, top: Swift.Bool, offset: CoreGraphics.CGFloat, round: CoreGraphics.CGFloat, bgcolor: Swift.Int, dismiss: Swift.Bool, cancelled: (() -> ())?) -> () in SASContractSDK(Overlay.o)
      "enum case for SwiftEntryKit.EKAttributes.RoundCorners.top(SwiftEntryKit.EKAttributes.RoundCorners.Type) -> (CoreGraphics.CGFloat) -> SwiftEntryKit.EKAttributes.RoundCorners", referenced from:
          function signature specialization <Arg[7] = Dead> of static SASContractSDK.Overlay.entrySheet(_: __C.UIView, top: Swift.Bool, offset: CoreGraphics.CGFloat, round: CoreGraphics.CGFloat, bgcolor: Swift.Int, dismiss: Swift.Bool, cancelled: (() -> ())?) -> () in SASContractSDK(Overlay.o)
          function signature specialization <Arg[7] = Dead> of static SASContractSDK.Overlay.entrySheetHide(_: __C.UIView, top: Swift.Bool, offset: CoreGraphics.CGFloat, round: CoreGraphics.CGFloat, bgcolor: Swift.Int, dismiss: Swift.Bool, cancelled: (() -> ())?) -> () in SASContractSDK(Overlay.o)
      "enum case for SwiftEntryKit.EKAttributes.BackgroundStyle.color(SwiftEntryKit.EKAttributes.BackgroundStyle.Type) -> (SwiftEntryKit.EKColor) -> SwiftEntryKit.EKAttributes.BackgroundStyle", referenced from:
          function signature specialization <Arg[1] = Dead> of static SASContractSDK.Overlay.entryAlert(__C.UIView) -> () in SASContractSDK(Overlay.o)
          function signature specialization <Arg[7] = Dead> of static SASContractSDK.Overlay.entrySheet(_: __C.UIView, top: Swift.Bool, offset: CoreGraphics.CGFloat, round: CoreGraphics.CGFloat, bgcolor: Swift.Int, dismiss: Swift.Bool, cancelled: (() -> ())?) -> () in SASContractSDK(Overlay.o)
          function signature specialization <Arg[7] = Dead> of static SASContractSDK.Overlay.entrySheetHide(_: __C.UIView, top: Swift.Bool, offset: CoreGraphics.CGFloat, round: CoreGraphics.CGFloat, bgcolor: Swift.Int, dismiss: Swift.Bool, cancelled: (() -> ())?) -> () in SASContractSDK(Overlay.o)
          function signature specialization <Arg[2] = Dead> of static SASContractSDK.Overlay.entryAlert(_: __C.UIView, dismiss: Swift.Bool) -> () in SASContractSDK(Overlay.o)
      "enum case for SwiftEntryKit.EKAttributes.PositionConstraints.Edge.ratio(SwiftEntryKit.EKAttributes.PositionConstraints.Edge.Type) -> (CoreGraphics.CGFloat) -> SwiftEntryKit.EKAttributes.PositionConstraints.Edge", referenced from:
          function signature specialization <Arg[1] = Dead> of static SASContractSDK.Overlay.entryAlert(__C.UIView) -> () in SASContractSDK(Overlay.o)
          function signature specialization <Arg[7] = Dead> of static SASContractSDK.Overlay.entrySheet(_: __C.UIView, top: Swift.Bool, offset: CoreGraphics.CGFloat, round: CoreGraphics.CGFloat, bgcolor: Swift.Int, dismiss: Swift.Bool, cancelled: (() -> ())?) -> () in SASContractSDK(Overlay.o)
          function signature specialization <Arg[7] = Dead> of static SASContractSDK.Overlay.entrySheetHide(_: __C.UIView, top: Swift.Bool, offset: CoreGraphics.CGFloat, round: CoreGraphics.CGFloat, bgcolor: Swift.Int, dismiss: Swift.Bool, cancelled: (() -> ())?) -> () in SASContractSDK(Overlay.o)
          function signature specialization <Arg[2] = Dead> of static SASContractSDK.Overlay.entryAlert(_: __C.UIView, dismiss: Swift.Bool) -> () in SASContractSDK(Overlay.o)
      "enum case for SwiftEntryKit.SwiftEntryKit.RollbackWindow.main(SwiftEntryKit.SwiftEntryKit.RollbackWindow.Type) -> SwiftEntryKit.SwiftEntryKit.RollbackWindow", referenced from:
          function signature specialization <Arg[1] = Dead> of static SASContractSDK.Overlay.entryAlert(__C.UIView) -> () in SASContractSDK(Overlay.o)
          function signature specialization <Arg[7] = Dead> of static SASContractSDK.Overlay.entrySheet(_: __C.UIView, top: Swift.Bool, offset: CoreGraphics.CGFloat, round: CoreGraphics.CGFloat, bgcolor: Swift.Int, dismiss: Swift.Bool, cancelled: (() -> ())?) -> () in SASContractSDK(Overlay.o)
          function signature specialization <Arg[7] = Dead> of static SASContractSDK.Overlay.entrySheetHide(_: __C.UIView, top: Swift.Bool, offset: CoreGraphics.CGFloat, round: CoreGraphics.CGFloat, bgcolor: Swift.Int, dismiss: Swift.Bool, cancelled: (() -> ())?) -> () in SASContractSDK(Overlay.o)
          function signature specialization <Arg[2] = Dead> of static SASContractSDK.Overlay.entryAlert(_: __C.UIView, dismiss: Swift.Bool) -> () in SASContractSDK(Overlay.o)
      "enum case for SwiftEntryKit.EKAttributes.Scroll.disabled(SwiftEntryKit.EKAttributes.Scroll.Type) -> SwiftEntryKit.EKAttributes.Scroll", referenced from:
          function signature specialization <Arg[1] = Dead> of static SASContractSDK.Overlay.entryAlert(__C.UIView) -> () in SASContractSDK(Overlay.o)
          function signature specialization <Arg[7] = Dead> of static SASContractSDK.Overlay.entrySheet(_: __C.UIView, top: Swift.Bool, offset: CoreGraphics.CGFloat, round: CoreGraphics.CGFloat, bgcolor: Swift.Int, dismiss: Swift.Bool, cancelled: (() -> ())?) -> () in SASContractSDK(Overlay.o)
          function signature specialization <Arg[2] = Dead> of static SASContractSDK.Overlay.entryAlert(_: __C.UIView, dismiss: Swift.Bool) -> () in SASContractSDK(Overlay.o)
      "enum case for SwiftEntryKit.EKAttributes.Animation.Translate.AnchorPosition.automatic(SwiftEntryKit.EKAttributes.Animation.Translate.AnchorPosition.Type) -> SwiftEntryKit.EKAttributes.Animation.Translate.AnchorPosition", referenced from:
          function signature specialization <Arg[1] = Dead> of static SASContractSDK.Overlay.entryAlert(__C.UIView) -> () in SASContractSDK(Overlay.o)
          function signature specialization <Arg[7] = Dead> of static SASContractSDK.Overlay.entrySheet(_: __C.UIView, top: Swift.Bool, offset: CoreGraphics.CGFloat, round: CoreGraphics.CGFloat, bgcolor: Swift.Int, dismiss: Swift.Bool, cancelled: (() -> ())?) -> () in SASContractSDK(Overlay.o)
          function signature specialization <Arg[7] = Dead> of static SASContractSDK.Overlay.entrySheetHide(_: __C.UIView, top: Swift.Bool, offset: CoreGraphics.CGFloat, round: CoreGraphics.CGFloat, bgcolor: Swift.Int, dismiss: Swift.Bool, cancelled: (() -> ())?) -> () in SASContractSDK(Overlay.o)
          function signature specialization <Arg[2] = Dead> of static SASContractSDK.Overlay.entryAlert(_: __C.UIView, dismiss: Swift.Bool) -> () in SASContractSDK(Overlay.o)
      "enum case for SwiftEntryKit.EKAttributes.Position.center(SwiftEntryKit.EKAttributes.Position.Type) -> SwiftEntryKit.EKAttributes.Position", referenced from:
          function signature specialization <Arg[1] = Dead> of static SASContractSDK.Overlay.entryAlert(__C.UIView) -> () in SASContractSDK(Overlay.o)
          function signature specialization <Arg[2] = Dead> of static SASContractSDK.Overlay.entryAlert(_: __C.UIView, dismiss: Swift.Bool) -> () in SASContractSDK(Overlay.o)
      "enum case for SwiftEntryKit.EKAttributes.PositionConstraints.Edge.intrinsic(SwiftEntryKit.EKAttributes.PositionConstraints.Edge.Type) -> SwiftEntryKit.EKAttributes.PositionConstraints.Edge", referenced from:
          function signature specialization <Arg[1] = Dead> of static SASContractSDK.Overlay.entryAlert(__C.UIView) -> () in SASContractSDK(Overlay.o)
          function signature specialization <Arg[7] = Dead> of static SASContractSDK.Overlay.entrySheet(_: __C.UIView, top: Swift.Bool, offset: CoreGraphics.CGFloat, round: CoreGraphics.CGFloat, bgcolor: Swift.Int, dismiss: Swift.Bool, cancelled: (() -> ())?) -> () in SASContractSDK(Overlay.o)
          function signature specialization <Arg[7] = Dead> of static SASContractSDK.Overlay.entrySheetHide(_: __C.UIView, top: Swift.Bool, offset: CoreGraphics.CGFloat, round: CoreGraphics.CGFloat, bgcolor: Swift.Int, dismiss: Swift.Bool, cancelled: (() -> ())?) -> () in SASContractSDK(Overlay.o)
          function signature specialization <Arg[2] = Dead> of static SASContractSDK.Overlay.entryAlert(_: __C.UIView, dismiss: Swift.Bool) -> () in SASContractSDK(Overlay.o)
      "enum case for SwiftEntryKit.SwiftEntryKit.EntryDismissalDescriptor.all(SwiftEntryKit.SwiftEntryKit.EntryDismissalDescriptor.Type) -> SwiftEntryKit.SwiftEntryKit.EntryDismissalDescriptor", referenced from:
          static SASContractSDK.Overlay.entryHide((() -> ())?) -> () in SASContractSDK(Overlay.o)
          @objc static SASContractSDK.Overlay.entryHide((() -> ())?) -> () in SASContractSDK(Overlay.o)
    ld: symbol(s) not found for architecture x86_64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)

```

### Stack

```
   CocoaPods : 1.11.3
        Ruby : ruby 3.0.0p0 (2020-12-25 revision 95aff21468) [x86_64-darwin21]
    RubyGems : 3.2.3
        Host : macOS 12.5.1 (21G83)
       Xcode : 13.4.1 (13F100)
         Git : git version 2.32.1 (Apple Git-133)
Ruby lib dir : /Users/adamshi/.rvm/rubies/ruby-3.0.0/lib
Repositories : 197749-saas-saas-ios-sasmodules-sasspecs - git - http://git.197749.com/saas/saas-ios/sasmodules/sasspecs.git @ 69605b7e007f43bdf300d85b9ad53ddb8c1ff5c7

               adamshi - git - https://github.com/AdamShi/FSCalendar.git @ 5ee1116c4818406a6d75458d14e77b616008a909

               aliyun-ptbpods - git - git@code.aliyun.com:pintuibao-ios/PTBPods.git @ 66e45535c29d21cd43524a7bb21a32ae80e0ae4d

               aliyun-saaspods - git - git@codeup.aliyun.com:60143076640593207be329d8/saas_framework/SAASPods.git @ a1a6745e93c3e8466430da6e1d77897258a01d6d

               aliyun-xsdpods - git - git@code.aliyun.com:pintuibao-ios/XSDPods.git @ d2d4e65481986f7dfb8315564979a03ac23f4d9f

               cocoapods - git - https://github.com/CocoaPods/Specs.git @ acb562291715033f20a930ca8e303e0703e4201c

               elfsundae-cocoapods-specs - git - https://github.com/ElfSundae/CocoaPods-Specs.git @ f0b535b352f6d4c712c5e01914d5981149ba9c2a

               JLPods - git - git@codeup.aliyun.com:600fb781841cc46b7c499921/jvlangyinqing_ios/JLPods.git @ 465f769f926a6e3dffe4560cd675d7cc0f2661dd

               SAASSpecs - git - http://git.197749.com/saasmodules/saasspecs.git @ fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
HEAD

               trunk - CDN - https://cdn.cocoapods.org/
               XCPods - git - git@codeup.aliyun.com:6010c87e841cc46b7c499ad3/xianchangyun/XCPods.git @ fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
HEAD

               YLSpecRep - git - git@iZm5e2qvb8hl5w1gjowpsxZ:YLTicketAPP_iOS_Group/YLSpecRep.git @ 0d22b02cc218249eecfa5f2589418a8891e68fe1

```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.5
cocoapods-packager    : 1.5.0
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.1
cocoapods-trunk       : 1.6.0
cocoapods-try         : 1.2.0
cocoapods-xcframework : 0.2.4
```

## Project that demonstrates the issue

[https://github.com/AdamShi/SASContractSDK](https://github.com/AdamShi/SASContractSDK)

**Comment by xuzhongping** [comment]
Same [issues](https://github.com/CocoaPods/CocoaPods/issues/11153), and I have an idea to solve it: in the dependency graph of pods, if a Pod is in Swift binary form, we need to set BUILD_LIBRARY_FOR_DISTRIBUTION to YES for all Pods downstream of its dependency.

**Comment by AdamShi** [comment]
> Same [issues](https://github.com/CocoaPods/CocoaPods/issues/11153), and I have an idea to solve it: in the dependency graph of pods, if a Pod is in Swift binary form, we need to set BUILD_LIBRARY_FOR_DISTRIBUTION to YES for all Pods downstream of its dependency.

    I have solved this problem. The reason is that I set `s.static_framework = true` in the podspec file, which generates a static library, so the verification can only be passed by adding `--skip-import-validation`; so finally I Deleted `s.static_framework = true`, no longer use the static library, but use the dynamic library, you don't need `--skip-import-validation`, you can pass the verification directly.
    To sum up, it is best to generate a dynamic library, and you can pass the verification without adding `--skip-import-validation`; if it is a static library, you need `--skip-import-validation` to pass the verification

**Comment by AdamShi** [comment]
I have solved this problem. The reason is that I set `s.static_framework = true` in the podspec file, which generates a static library, so the verification can only be passed by adding `--skip-import-validation`; so finally I Deleted `s.static_framework = true`, no longer use the static library, but use the dynamic library, you don't need `--skip-import-validation`, you can pass the verification directly.
To sum up, it is best to generate a dynamic library, and you can pass the verification without adding `--skip-import-validation`; if it is a static library, you need `--skip-import-validation` to pass the verification

**Comment by AdamShi** [comment]
> Same [issues](https://github.com/CocoaPods/CocoaPods/issues/11153), and I have an idea to solve it: in the dependency graph of pods, if a Pod is in Swift binary form, we need to set BUILD_LIBRARY_FOR_DISTRIBUTION to YES for all Pods downstream of its dependency.

I have solved this problem. The reason is that I set `s.static_framework = true` in the podspec file, which generates a static library, so the verification can only be passed by adding `--skip-import-validation`; so finally I Deleted `s.static_framework = true`, no longer use the static library, but use the dynamic library, you don't need `--skip-import-validation`, you can pass the verification directly.
To sum up, it is best to generate a dynamic library, and you can pass the verification without adding `--skip-import-validation`; if it is a static library, you need `--skip-import-validation` to pass the verification
